x-pack/plugins/maps/public/embeddable/map_embeddable.js

```diff
@@ -34,31 +34,21 @@ import {
 } from '../store/ui';
 import { getInspectorAdapters } from '../store/non_serializable_instances';
 import { getMapCenter, getMapZoom } from '../selectors/map_selectors';
-import { i18n } from '@kbn/i18n';
+import { MAP_SAVED_OBJECT_TYPE } from '../../common/constants';
 
 export class MapEmbeddable extends Embeddable {
+  type = MAP_SAVED_OBJECT_TYPE;
 
-  constructor({
-    onEmbeddableStateChanged,
-    embeddableConfig,
-    savedMap,
-    editUrl,
-    editable,
-    indexPatterns = []
-  }) {
-    super({
-      title: savedMap.title,
-      editUrl,
-      editLabel: i18n.translate('xpack.maps.embeddable.editLabel', {
-        defaultMessage: 'Edit map',
-      }),
-      editable,
-      indexPatterns });
-
-    this._onEmbeddableStateChanged = onEmbeddableStateChanged;
-    this._embeddableConfig = _.cloneDeep(embeddableConfig);
-    this._savedMap = savedMap;
+  constructor(config, initialInput, parent) {
+    super(
+      initialInput,
+      { editUrl: config.editUrl, indexPatterns: config.indexPatterns, editable: config.editable, defaultTitle: config.savedMap.title },
```

Comment:
nit: extract or split into multiple lines

Reply:
if only this was a typescript file it'd do it for me! 😬 :)